crates/ty_python_semantic/src/types/class.rs

```diff
@@ -3653,7 +3656,8 @@ impl KnownClass {
             | Self::Super
             | Self::GenericAlias
             | Self::Deque
-            | Self::Bytes => true,
+            | Self::Bytes
+            | Self::Template => true,
```

Comment:
`Template` is [marked as `@final` in typeshed](https://github.com/python/typeshed/blob/91e2ed0953592795fd8c29e3005a1315bf652ffc/stdlib/string/templatelib.pyi#L7-L8), so I don't think we also need to consider it a disjoint base (see the comment a couple of lines below this one) -- it'll just result in us emitting two diagnostics if you tried to inherit from it, rather than one diagnostic. One seems enough :-)